include/mbgl/storage/file_source.hpp

```diff
@@ -35,24 +36,28 @@ enum FileSourceType : uint8_t {
 // GeoJSONSource, RasterSource, VectorSource, CustomGeometrySource and other *Sources.
 class FileSource {
 public:
+    template <typename T = void()>
+    using Callback = std23::move_only_function<T>;
+    template <typename T = void()>
+    using CopyableCallback = std::function<T>;
```

Comment:
I personally don't like these at all. I think it's much cleaner to have APIs don't obfuscate the fact that they use simple STL templates.